vulture_os/toolkit/api_parser/beyondtrust_reportings/beyondtrust_reportings.py

```diff
@@ -197,22 +198,47 @@ def format_team_logs(self, logs):
 
         return formated_logs
 
+    def format_event_value(self, event):
+        child_session_details = deepcopy(event)
+
+        if 'data' in event:
+            data = {}
+            event_value = event['data']['value']
+            logger.debug(event_value, extra={'frontend': str(self.frontend)})
+
+            if isinstance(event_value, list):
+                for elem in event_value:
+                    data[elem['@name']] = elem['@value']
+            elif isinstance(event_value, dict):
+                data[event_value['@name']] = event_value['@value']
+            child_session_details['data'] = data
+
+        return child_session_details
+
     def format_access_session_logs(self, logs):
-        formated_logs = []
         try:
+            formated_logs = []
             sessions_logs = logs['session_list'].get('session', [])
             if not isinstance(sessions_logs, list):
                 sessions_logs = [sessions_logs]
 
             for log in sessions_logs:
-                del log['session_details']  # Avoid too long session log
                 log["@timestamp"] = log["start_time"]["@timestamp"]
-                formated_logs.append(log)
+
+                if not log.get('session_details', {}).get('event'):
+                    formated_logs.append(log)
+                    continue
+
+                for event in log['session_details']['event']:
+                    child_log = deepcopy(log)
+                    child_log['session_details'] = self.format_event_value(event)
+                    formated_logs.append(child_log)
+
         except KeyError as e:
-            logger.error(f"[{__parser__}] An error occurred while formating AccessSession logs: {e}",
+            logger.error(f"[{__parser__}] An error occurred while formating AccountActivityList logs: {e}",
                         extra={'frontend': str(self.frontend)})
-
-        return formated_logs
+        else:
+            return formated_logs
```

Comment:
Now. if there's a KeyError the function will return None, is that what you want ?